src/test/testRunner.ts

```diff
@@ -1,32 +1,256 @@
+        }
+
+        // TODO consider putting this under a conditional flag
+        // Files that are not touched by code ran by the test runner is manually instrumented, to
+        // illustrate the missing coverage.
+        // @ts-ignore - Implicit any
+        self.matchFn.files.forEach(file => {
+            try {
+                if (!cov[file]) {
+                    console.log(file)
+                    self.transformer(fs.readFileSync(file, 'utf-8'), file)
+
+                    // When instrumenting the code, istanbul will give each FunctionDeclaration a value of 1 in coverState.s,
+                    // presumably to compensate for function hoisting. We need to reset this, as the function was not hoisted,
+                    // as it was never loaded.
+                    Object.keys(self.instrumenter.coverState.s).forEach(key => {
+                        self.instrumenter.coverState.s[key] = 0
+                    })
+
+                    cov[file] = self.instrumenter.coverState
+                }
+            } catch (e) {
+                console.log(e)
+            }
+        })
+
+        // TODO Allow config of reporting directory with
+        let reportingDir = path.join(self.testsRoot, self.options.relativeCoverageDir)
+        let includePid = self.options.includePid
+        let pidExt = includePid ? '-' + process.pid : ''
+        let coverageFile = path.resolve(reportingDir, 'coverage' + pidExt + '.json')
+
+        _mkDirIfExists(reportingDir) // yes, do this again since some test runners could clean the dir initially created
+
+        fs.writeFileSync(coverageFile, JSON.stringify(cov), 'utf8')
+
+        let remappedCollector = remapIstanbul.remap(cov, {
+            // @ts-ignore - Implicit any
+            warn: warning => {
+                // We expect some warnings as any JS file without a typescript mapping will cause this.
+                // By default, we"ll skip printing these to the console as it clutters it up
+                if (self.options.verbose) {
+                    console.warn(warning)
+                }
+            },
+        })
+
+        let reporter = new istanbul.Reporter(undefined, reportingDir)
+        let reportTypes = self.options.reports instanceof Array ? self.options.reports : ['lcov']
+        reporter.addAll(reportTypes)
+        reporter.write(remappedCollector, true, () => {
+            console.log(`Code coverage reports written to ${reportingDir}`)
+        })
+    }
+}
+
+function _mkDirIfExists(dir: string): void {
+    if (!fs.existsSync(dir)) {
+        fs.mkdirSync(dir)
+    }
+}
+
+interface TestRunnerOptions {
+    enabled?: boolean
+    relativeCoverageDir: string
+    relativeSourcePath: string
+    ignorePatterns: string[]
+    includePid?: boolean
+    reports?: string[]
+    verbose?: boolean
+}
+
+function _readCoverOptions(testsRoot: string): TestRunnerOptions {
+    let coverConfigPath = path.join(testsRoot, '..', '..', 'coverconfig.json')
```

Comment:
We were previously running from the `dist/src` directory. With this change, we are now running from the `dist` directory.

This was the root for the VSCode sample. I left it because it makes more sense since it emulates the packaged path.

packaged : `<root>/package.json`, `<root>/extensionMain.js`, `<root>/dist/extension.js`

test: `<root>/dist/package.json`, `<root>/dist/extensionMain.js`, `<root>/dist/src/extension.js`

The difference being that stuff in src gets webpacked and put directly into `dist`.

I'm pretty certain this could be made better, but this appears to work properly